src/components/Chatbox.tsx

```diff
@@ -48,46 +51,71 @@ const Chatbox: FC<Props> = ({
   isLoading,
   chatId,
   showHeader = false,
+  showAdminTools = false,
   lang = langs.en,
   currentMember,
   members,
 }) => {
   const useStyles = makeStyles((theme) => ({
     container: {
-      height: height || DEFAULT_CHATBOX_HEIGHT,
+      display: 'flex',
+      flexDirection: 'column',
       padding: theme.spacing(0, 1),
+      height: height || DEFAULT_CHATBOX_HEIGHT,
+    },
+    bottomContainer: {
+      boxSizing: 'border-box',
+      paddingBottom: theme.spacing(1),
     },
   }));
   const classes = useStyles();
-  const i18n = buildI18n(namespaces.chatbox);
-  i18n.changeLanguage(lang);
+  const i18n = useMemo(() => {
+    const i18nInstance = buildI18n(namespaces.chatbox);
+    i18nInstance.changeLanguage(lang);
+    return i18nInstance;
+  }, [lang]);
+  const ref = useRef<HTMLDivElement>(null);
+  const [inputBarHeight, setInputBarHeight] = useState(0);
+
+  useEffect(() => {
+    setInputBarHeight(ref.current?.clientHeight || 0);
+  }, [showAdminTools, ref]);
 
   if (isLoading) {
     return null;
   }
+
   return (
     <I18nextProvider i18n={i18n}>
       <EditingContextProvider>
         <HooksContextProvider useAvatarHook={useAvatarHook}>
-          <Fragment>
-            {showHeader && <Header />}
-            <Container id={id} maxWidth="md" className={classes.container}>
-              <Messages
-                members={members}
-                currentMember={currentMember}
-                messages={messages}
-                height={height - INPUT_HEIGHT}
-                deleteMessageFunction={deleteMessageFunction}
-                editMessageFunction={editMessageFunction}
-              />
-              <InputBar
-                chatId={chatId}
-                sendMessageBoxId={sendMessageBoxId}
-                sendMessageFunction={sendMessageFunction}
-                editMessageFunction={editMessageFunction}
-              />
-            </Container>
-          </Fragment>
+          <MessagesContextProvider
+            messages={messages}
+            chatId={chatId}
+            members={members}
+          >
+            <>
+              {showHeader && <Header />}
+              <Container id={id} maxWidth="md" className={classes.container}>
+                <Messages
+                  members={members}
+                  currentMember={currentMember}
+                  messages={messages}
```

Comment:
It's up to you, but it might be interesting for `Messages` to get `messages` and `members` from `MessagesContext` directly.

Reply:
Yes this is a good idea, i also changed other components to use this context.